### DISCUSSION TITLE
S40 New 4.25 Firmware Available

### DISCUSSION THREAD

**Discussion by 42Network** [PeteRager/lennoxs30#262] [General]
Perhaps some good news: just got an email from Lennox, quoted below, mentioning a few features, one relates to temporary diagnostic mode. Not sure if there are any bug fixes. 

Is there anything special I should to to help with HA integration troubleshooting prior to letting my S40 upgrade?

And in my particular case, should I delete my Lennox integration first to delete my old remote sensors?

Thanks

Lennox 

 
S40 & Service Dashboard Features: Equipment Configuration
Profiles & Temporary Advanced Diagnostics
New time-saving features are now available on the Lennox S40 Smart Thermostat through the use of Service Dashboard. The latest software update (4.25) includes new Equipment Configuration Profiles and Temporary Advanced Diagnostics, allowing for fast and efficient solutions when installing and servicing Lennox equipment.

Equipment Configuration Profiles (ECP)
ECP is a feature designed to help your technicians achieve higher quality and faster installations. This new feature allows you to create unique equipment profiles specific to products sold at your dealership and equipment profiles tailored to your local weather conditions from the comfort of your desktop or phone. These profiles are created through a step-by-step setup flow available on Service Dashboard. Through the Smart Tech App, technicians can scan the equipment bar code and apply the associated equipment profile for an easier configuration process.

Temporary Advanced Diagnostics
With the use of Service Dashboard, Temporary Advanced Diagnostics allows you to send push notifications to homeowners via text, asking them to provide you with temporary 24-hour access to view their system, diagnose, troubleshoot, and fix problems remotely. This proactive care helps eliminate downtime and avoid unnecessary service visits and repair costs, ensuring nothing short of perfect air.
Visit Service Dashboard

Follow us on social media

**Comment by 42Network** [comment]
To answer my own discussion question, I just removed my Lennox HA integration, upgraded my S40 to this new 4.25 firmware, and reinstalled the integration. It was discovered successfully, and of course I no longer have the phantom remote sensors anymore.

**Comment by PeteRager** [comment]
Apologies for the delays, away on holiday, glad you got it sorted out.

**Comment by JHenares** [comment]
I am new here.

I Need help with my HA and Lennox S40 integration.

 I have HA and Lennox S40 thermostat.

My Lennox system has 4 zones. I had problems with my Lennox since it was installed because my old HVAC ducts were reused and restricted.

To resolve my issues with the s250 errors (The furnace primary limit switch is open), I have installed inline booster fans (AC Infinity) to zones that are restricted. My automation interrogates if there is a heating demand for the restricted zone and turn on the booster fan. It worked and has been working for more than a year now and suddenly it stopped working. It looks like my Lennox S40 thermostat had a software update last Nov 5, 2024 and I am now in software revision 4.31.0582. My fault because I left the default of auto update to ON. 

My trigger looks like: 
entity_id:
  - climate.pb_sd_s40_office
attribute: hvac_action
from: heating
trigger: state

Now, the trigger in my automation has this error "Error in describing trigger: Cannot read properties of undefined (reading 'entity_id')" to the trigger.

If I look at the "STATES", I can only see temperature of each zone. No hvac_action, zoneEnabled, zoningMode... No state_attr current_temperature, target_temp_low..

Is this because of software revision 4.31.0582? Does anyone else have this problem? What is the solution? Do I need to replace my S40 thermostat hoping that it would have an older software revision?

Thank you very much in advance.

**Reply by PeteRager** [reply]
Here are a couple things to do.

First, rather than using this discussion please create a new issue / bug report here - https://github.com/PeteRager/lennoxs30/issues and attach the information requested below:

In the integration configuration - what devices do you see? In developer tools /state - what climate entities do you see?  Attach some screen shots. In devices you should see an entry for each thermostat.

I'd try power cycling the Lennox system.  Shutdown HA. Cut power to the S40s if you can and wait for them to reboot. Alternatively go to the panel and there is an options to restart the hub.  Then Restart HA.

If things still look wrong. Enable debug logging and message logging as described here: https://github.com/PeteRager/lennoxs30?tab=readme-ov-file#reporting-bugs ; restart HA; let it run for a few minutes and then grab the message log and home-assistant.log and attach in the new issue.

**Comment by JHenares** [comment]
Hi Pete,

First of all, thank you for replying to my posting and sorry for posting in Discussion #262.

I will create a new issue/bug report.

I will add in the  "new issue/bug report" all requested items.

I will also try power cycling and if it does not fix the issues, enable debug logging and attach files to "new issue/bug report".

Thank you again.

John Henares



I will power cycle to check if it resolves the issue

**Comment by JHenares** [comment]
Hi Pete,

I tried power cycling and it did not resolve the issue.

I was ready to create a new issue and went to the Lennox S30 Integration to turn enable/disable debugging and I found an option to update the Lennox s30 integration. I applied all updates and that resolved the issue.

Thanks you very much.

John
